taskpools/primitives/barriers_posix.nim

```diff
@@ -31,15 +35,15 @@ when defined(osx):
   export pthread_barrier_init, pthread_barrier_wait, pthread_barrier_destroy
 else:
   proc pthread_barrier_init*(
-        barrier: PthreadBarrier,
-        attr: PthreadAttr or ptr PthreadAttr,
+        barrier: var PthreadBarrier,
+        attr: ptr PthreadAttr,
         count: range[0'i32..high(int32)]
       ): Errno {.header: "<pthread.h>".}
     ## Initialize `barrier` with the attributes `attr`.
     ## The barrier is opened when `count` waiters arrived.
 
   proc pthread_barrier_destroy*(
-        barrier: sink PthreadBarrier): Errno {.header: "<pthread.h>".}
+        barrier: sink var PthreadBarrier): Errno {.header: "<pthread.h>".}
```

Comment:
`sink` implies `var`
The rational is that if an object is `sinked` there is a single owner and no reader afterwards so it can be mutated, in particular to destroy it.

Reply:
indeed - this was mostly an attempt to get Nim to pass it by pointer to C

also, in general, in 1.2, we're not really ready for `sink` - it's pure luck that it has worked so far.